composer/loggers/progress_bar_logger.py

```diff
@@ -123,47 +141,65 @@ def should_log(state: State, log_level: LogLevel, console_log_level: LogLevel =
             self.should_log = should_log
         else:
             self.should_log = console_log_level
+
+        # set the stream
         if isinstance(stream, str):
-            if stream.lower() == 'stdout':
-                stream = sys.stdout
-            elif stream.lower() == 'stderr':
-                stream = sys.stderr
-            else:
-                raise ValueError(
-                    f"Invalid stream option: Should be 'stdout', 'stderr', or a TextIO-like object; got {stream}")
+            stream = {
+                'stdout': sys.stdout,
+                'stderr': sys.stderr,
+            }[stream.lower()]
```

Comment:
nit: Would prefer that a valueerror is raised on an invalid stream option (I agree with your comment: https://mosaicml.slack.com/archives/C0205HZTT9P/p1655512719555359?thread_ts=1655172522.700169&cid=C0205HZTT9P)